site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: false
+title: State management
+overview: In this tutorial, you will learn what a Containers and Injectors are and how to use them to manage external state and inject this into the parts of the widget tree.
+---
+
+{% section 'first' %}
+
+# State management
+
+## Overview
+
+Almost all non-trivial applications have multiple states that they can exist in. While Dojo 2's widgets can manage application state, they can quickly become overburdened as they manage their own visual representations, listen for interactions from the user, manage their children, and keep track of state information. Additionally, using widgets to pass state through an application often forces the widgets to be aware of state information for the sole purpose of passing that data down to their children. To allow widgets to remain focused on their primary roles of providing a visual representation of themselves and listening for user interactions, Dojo 2 provides `Containers` and `Injectors` that are designed to manage an application's state and provide that information only to the widgets that require it.
+
+In this tutorial, we will start with an application that is using widgets to manage its state. We will then extract all of the state-related code out of the widgets and inject it into the widgets that need it. You can [download](../assets/1010_form_widgets-initial.zip) the demo project to get started.
+
+## Prerequisites
+
+This tutorial assumes that you have gone through the [beginner tutorial](../001_static_content) series.
+
+{% section %}
+
+## Creating an application context
+
+{% task 'Create a class to manage application state.' %}
+
+To begin our discussion, let's take a look at the initial version of `App`:
+
+{% include_codefile 'demo/initial/biz-e-corp/src/widgets/App.ts' %}
+
+Most of this widget is dedicated to holding and managing the `WorkerData` in the application. Notice, however, that it never actually uses that data itself. `App` is only holding the data so that its children can access it. While this may be acceptable in a small application, in larger applications, the `App` class can become overwhelmed by the amount of state information that it is tracking. Since the state information not a primary concern of the `App` class, let's refactor it out of `App` and into a new class.
+
+{% instruction 'Add the following to `ApplicationContext.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' %}
+
+The code begins by importing some modules, including the `WorkerProperties` and `WorkerFormData` interfaces defined in the `Worker` and `WorkerForm` modules. These two interfaces define the state that the `ApplicationContext` is going to track. One important item to note is that the `ApplicationContext` is aware of the widgets, but the widgets are not aware of the `ApplicationContext` module. It is important to import the modules this way since it minimizes the widgets' dependencies and makes them easier to test.
```

Comment:
"minimizes the widgets' dependencies and makes them easier to test"  Perhaps "improves the testability of widgets by minimizing the widget dependency chain"... that said, it still feels a bit under-substantiated.